### DISCUSSION TITLE
Enhancing Conflict Resolution in Weblate with an 'Auto Resolve' Feature

### DISCUSSION THREAD

**Discussion by vanlinh2204** [WeblateOrg/weblate#10200] [Ideas]
Collaboration between developers and translators on Weblate can sometimes lead to conflicts when both parties update the same key. Currently, manual intervention or collaboration through Git is required to resolve such conflicts. In this discussion, let's explore the idea of introducing an "Auto Resolve" feature in Weblate, drawing inspiration from Git's --strategy-option ours/theirs. This could potentially simplify conflict resolution, making the workflow smoother for both developers and translators. Share your thoughts, experiences, and suggestions on how we can enhance collaboration efficiency in Weblate!

**Comment by nijel** [comment]
There are two kind of conflicts:

* The structure of the translation file changes, but there are no conflicts in the actual strings. These can be typically handled by an appropriate merge driver, Weblate currently does this for gettext PO by https://github.com/WeblateOrg/weblate/blob/main/weblate/examples/git-merge-gettext-po. Adding something like this for other formats is doable, just somebody needs to develop it.
* The actual conflict is in the translated string - it has been edited both in Weblate and externally. In this case somebody still has to review the changes and take one of them. There is https://github.com/WeblateOrg/weblate/issues/98 open for years, but I'm still not convinced this is a way to go.

There also is https://github.com/WeblateOrg/weblate/issues/7103 on the topic.
